tests/unit/test_parallel_write.py

```diff
@@ -0,0 +1,185 @@
+"""Module for testing the parallel write feature for the ZarrIO."""
+import unittest
+from pathlib import Path
+from typing import Tuple, Dict
+from io import StringIO
+from unittest.mock import patch
+
+import numpy as np
+from numpy.testing import assert_array_equal
+from hdmf_zarr import ZarrIO
+from hdmf.common import DynamicTable, VectorData, get_manager
+from hdmf.data_utils import GenericDataChunkIterator, DataChunkIterator
+
+try:
+    import tqdm  # noqa: F401
+    TQDM_INSTALLED = True
+except ImportError:
+    TQDM_INSTALLED = False
```

Comment:
HDMF has https://github.com/hdmf-dev/hdmf/blob/dev/requirements-opt.txt

but I see no such analog here - how should I proceed on adding an optional dependency such as `tqdm`?

Reply:
@rly thoughts. I think adding a similar `requirements-opt.txt` is fine.

Reply:
OK think I got them all, but there seems to be a fair number of places that need to be updated (CI calls; various hidden config files, etc) so someone should probably double-check

Reply:
Thanks. @rly could you review this PR and check that the requirements (in particular the new optional requirements) are correct and that we didn't miss anything.

Reply:
Sorry I missed these pings. I will review